### ISSUE TITLE
Fyne v2.4: Markdown only shows one horizontal rule

### ISSUE THREAD

**Issue by Klaus-Tockloth** [fyne-io/fyne#4216]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Markdown only shows one horizontal rule in Fyne v2.4.

### How to reproduce

Compile and run example code.

### Screenshots

Fyne 2.3.5:
<img width="752" alt="Bildschirmfoto 2023-09-04 um 09 25 56" src="https://github.com/fyne-io/fyne/assets/5493128/4556a194-856c-4296-b897-c01f90ed1073">

Fyne 2.4.0:
<img width="776" alt="Bildschirmfoto 2023-09-04 um 09 28 25" src="https://github.com/fyne-io/fyne/assets/5493128/b42275b5-a6cf-495f-9309-ac0e5da7ff7b">

### Example code

```go
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {

	a := app.New()
	w := a.NewWindow("Markdown Viewer")
	w.Resize(fyne.NewSize(700, 700))

	preview := widget.NewRichTextFromMarkdown(generalHelp)
	preview.Wrapping = fyne.TextWrapWord

	w.SetContent(container.NewScroll(preview))
	w.ShowAndRun()
}

var generalHelp = `**Was beinhaltet die LoremIpsun PKI-Karte?**

Die LoremIpsun PKI-Karte enthält zwei oder drei Sicherheitszertifikate mit einer begrenzten Lebensdauer.
Bevor die Sicherheitszertifikate ungültig werden, sind sie durch diesen Prozess zu erneuern.

---

**Wie erneuere ich die Sicherheitszertifikate auf der LoremIpsun PKI-Karte?**

Die Erneuerung der Sicherheitszertifikate erfolgt über diese Applikation. Oft sind am Rechner des Nutzers mehrere Kartenleser angeschlossen. 
Der typische Ablauf sieht deshalb folgendermaßen aus:
- Lesen aller Karten.
- Auswahl der Karte für die Zertifikate zu erneuern sind.
- Eingabe der Karten-PIN.
- Prüfung des Ablaufdatums.
- Start des Erneuerungsvorgangs.

---

**Ist die Erneuerung der Sicherheitszertifikate jederzeit möglich?**

Die Sicherheitszertifikate können nur in einem bestimmten Zeitraum vor und nach dem Ablaufdatum erneuert werden.

---

**Was ist beim Erneuerungsvorgang zu beachten?**

Im Rahmen der Erneuerung der Sicherheitszertifikate erfolgen etliche Lese- und Schreibzugriffe gegen die LoremIpsun PKI-Karte.
Während der Vorgang läuft, darf die Karte nicht entfernt werden.

---

**Was ist zu tun, wenn ein Problem auftritt?**

Mögliche Probleme und deren Ursachen sind in der Hilfesektion "Probleme" beschrieben.

---

**Wie kann ich prüfen, ob die Erneuerung der Sicherheitszertifikate erfolgreich war?**

Die Erfolgsmeldung am Ende des Erneuerungsprozesses weist das neue Ablaufdatum der Sicherheitszertifikate aus.
Das neue Ablaufdatum wird ebenfalls im Hauptfenster angezeigt.
`

```

### Fyne version

2.4.0

### Go compiler version

go 1.21

### Operating system and version

macOS 13.5.1

**Comment by andydotxyz** [comment]
Unfortunately this was caused by 82b567dd, it seems that the uniqueness is not understood by the `RichText` usage of `visualCache` (`map[RichTextSegment][]fyne.CanvasObject`) - what do you think @Jacalz should that commit be reverted?

**Comment by Jacalz** [comment]
Opened a fix now. Sorry for breaking this.

**Comment by dweymouth** [comment]
Is this fixed after #4217?

**Comment by Jacalz** [comment]
It seems like the commit was only ever merged to develop and not the release branch. It will have to be in v2.4.2 :/

**Comment by Jacalz** [comment]
Perhaps @andydotxyz can backport to the release channel?

**Comment by dweymouth** [comment]
I was just wondering if we should close this ticket since there's a fix on develop now

**Comment by Jacalz** [comment]
I'd say that we close it once the commit has been backported, otherwise we might miss to do so 😅

**Comment by andydotxyz** [comment]
I've picked it over now, and the others that have landed since 2.4.1
